git/repo/base.py

```diff
@@ -1012,6 +1012,9 @@ def clone_from(cls, url, to_path, progress=None, env=None, multi_options=None, *
         :param to_path: Path to which the repository should be cloned to
         :param progress: See 'git.remote.Remote.push'.
         :param env: Optional dictionary containing the desired environment variables.
+            Note: Provided variables will be used to update the execution
+            environment for `git`. If some variable is not specified in `env`
```

Comment:
Maybe you could also hint at how to 'unset' environment variables that way.
It's just a suggestion though, I am happy to merge as is.

Reply:
Sure, I'll mention that :)